internal/configs/virtualserver.go

```diff
@@ -284,6 +284,10 @@ func (vsc *virtualServerConfigurator) GenerateVirtualServerConfig(vsEx *VirtualS
 
 	// generate upstreams for VirtualServer
 	for _, u := range vsEx.VirtualServer.Spec.Upstreams {
+		if (vsEx.VirtualServer.Spec.TLS != nil || !vsc.cfgParams.HTTP2) && u.GRPC {
```

Comment:
there is a bug here - it should be `vsEx.VirtualServer.Spec.TLS == nil`

also, there are more cases to determine TLS termination than `vsEx.VirtualServer.Spec.TLS == nil`. Those are covered in `generateSSLConfig`. Because of that, we should use `sslConfig == nil` here

```suggestion
		if (sslConfig == nil || !vsc.cfgParams.HTTP2) && u.GRPC {
```

same for the VSR upstreams